qiskit_experiments/library/characterization/resonator_spectroscopy.py

```diff
@@ -0,0 +1,176 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Spectroscopy experiment class for resonators."""
+
+from typing import Iterable, Optional, Tuple
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.exceptions import QiskitError
+from qiskit.providers import Backend
+import qiskit.pulse as pulse
+
+from qiskit_experiments.library.characterization import QubitSpectroscopy
+from qiskit_experiments.data_processing.processor_library import get_processor, ProjectorType
+from .analysis.resonator_spectroscopy_analysis import ResonatorSpectroscopyAnalysis
+
+
+class ResonatorSpectroscopy(QubitSpectroscopy):
+    """Perform spectroscopy on the readout resonator.
+
+    # section: overview
+        This experiment does spectroscopy on the readout resonator. It applies the following
+        circuit
+
+        .. parsed-literal::
+
+                 ┌─┐
+              q: ┤M├
+                 └╥┘
+            c: 1/═╩═
+                  0
+
+        where a spectroscopy pulse is attached to the measurement instruction. When doing
+        readout resonator spectroscopy, each measured IQ point has a frequency dependent
+        phase. Close to the resonance, the IQ points start rotating around in the IQ plan.
+        To create a meaningful signal this experiment therefore uses a custom data processor
+        where the dimensionality reducing SVD is replaced by the absolute value of the IQ
+        point.
+    """
+
+    def __init__(
+        self,
+        qubit: int,
+        frequencies: Iterable[float],
+        backend: Optional[Backend] = None,
+        absolute: bool = True,
+    ):
+        """
+        A spectroscopy experiment run by setting the frequency of the readout drive.
+        The parameters of the GaussianSquare spectroscopy pulse can be specified at run-time.
+        The spectroscopy pulse has the following parameters:
+        - amp: The amplitude of the pulse must be between 0 and 1, the default is 0.1.
```

Comment:
The works, see here: 5515f2d. However, I recommend changing the name to experiment_options in the CR Hamiltonian tomography since we are only setting the experiment options in this fashion.